src/context/chains.ts

```diff
@@ -34,6 +43,25 @@ export async function assembleChainMetadata() {
     await registry.listRegistryContent();
   }
 
-  const chains = { ...defaultChainMetadata, ...customChainMetadata };
+  const chains: ChainMap<ChainMetadata> = Object.entries({
+    ...defaultChainMetadata,
+    ...customChainMetadata,
+  }).reduce((accum, [name, chain]) => {
+    const privateRpc = rpcOverrides.success ? rpcOverrides.data[name] : null;
+    if (privateRpc) {
+      return {
+        ...accum,
+        [name]: {
+          ...chain,
+          rpurls: chain.rpcUrls,
```

Comment:
Typo `rpurls`. But also the type looks wrong. Above and in your example it looks like rpcOverrides is a map of chain name -> rpcUrlSchema. However the rpcUrls field is an array.

Reply:
Yep this was half baked, apologies. Now type safe